src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -435,7 +435,33 @@ namespace winrt::TerminalApp::implementation
     void CommandPalette::_rootPointerPressed(Windows::Foundation::IInspectable const& /*sender*/,
                                              Windows::UI::Xaml::Input::PointerRoutedEventArgs const& /*e*/)
     {
-        _dismissPalette();
+        if (Visibility() != Visibility::Collapsed)
+        {
+            _dismissPalette();
+        }
+    }
+
+    // Method Description:
+    // - The purpose of this event handler is to hind the palette if it loses focus.
+    // Since the focus might move within the palette we need to register this handler for all focusable children
+    // Right now this is achieved by explicitly registering on the root on the highlighted text controls.
+    // Few notes:
+    // - found no way to entires disable focus on list-view elements in single-item-selection mode without breaking list behavior
+    // - TODO: the explicit implementation can be generalized by checking if the focused element is a descendant of palette
```

Comment:
Do we want to address this TODO? Are there any drawbacks from the current approach? Like, will he manually have to add this `_lostFocusHandler` to all the focusable UI elements in the cmdpal?

Reply:
The drawback of the current approach is that we add another field we need to remember to add it here.

Reply:
Okay. That's definitely annoying to remember, but hopefully we'll remember to do this.

Reply:
> Okay. That's definitely annoying to remember, but hopefully we'll remember to do this.

IT IS NOW FIXED HA HA HA